src/engine/blocks.js

```diff
@@ -149,17 +168,22 @@ class Blocks {
      * @return {?string} ID of procedure definition.
      */
     getProcedureDefinition (name) {
+        if (this._procedureDefinitions.hasOwnProperty(name)) {
+            return this._procedureDefinitions[name];
+        }
         for (const id in this._blocks) {
             if (!this._blocks.hasOwnProperty(id)) continue;
             const block = this._blocks[id];
             if (block.opcode === 'procedures_defnoreturn' ||
                 block.opcode === 'procedures_defreturn') {
                 const internal = this._getCustomBlockInternal(block);
                 if (internal && internal.mutation.proccode === name) {
-                    return id; // The outer define block id
+                    this._procedureDefinitions[name] = id; // The outer define block id
+                    return id;
                 }
             }
         }
+        this._procedureDefinitions[name] = null; // This is not the same as undefined
```

Comment:
`hasOwnProperty` doesn't test if `[name]` is `undefined`. Its equivalent logically to `Object.keys(this._procedureDefinitions).indexOf(name) > -1`. You could set this to `undefined` and `hasOwnProperty` will return `true`.